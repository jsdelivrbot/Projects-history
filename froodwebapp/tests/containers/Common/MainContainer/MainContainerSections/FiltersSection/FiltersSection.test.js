import FilterTabs from 'containers/Common/MainContainer/MainContainerSections/FiltersSection/FiltersSection';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('FilterTabs', () => {
  const props = {
    visible: false,
    limit: 10,
    offset: 15,
    activeFilterId: '1',
    searchSectionFilters: [],
    filters: [],
    handleUpdateDefaultColumns: jest.fn(),
    columns: [{
      id: 1,
      order: 'first order',
      displayName: 'Inventory',
      isDefault: false
    }, {
      id: 2,
      order: 'second order',
      displayName: 'Price lists',
      isDefault: true
    }],
    updateSearchSectionFilters: jest.fn(),
    getWithFilterRequest: jest.fn(),
    saveFilterRequest: jest.fn(),
    updateFilterRequest: jest.fn(),
    deleteFilterRequest: jest.fn(),
    updateColumnsRequest: jest.fn(),
    getAllItemsRequest: jest.fn(),
    changeActiveFilterId: jest.fn()
  };

  it('renders the FilterTabs component', () => {
    const component = shallow(<FilterTabs { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of FilterTabs', () => {
    const component = shallow(<FilterTabs { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });

  it('calls handleToggleSearchSection and sets searchSectionVisible to state', () => {
    const component = shallow(<FilterTabs { ...props } />);
    component.instance().handleToggleSearchSection();

    expect(component.state('searchSectionVisible')).toEqual(true);
  });
});
