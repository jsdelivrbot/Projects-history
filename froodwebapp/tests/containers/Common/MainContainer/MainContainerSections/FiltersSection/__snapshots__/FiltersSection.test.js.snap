// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTabs capturing Snapshot of FilterTabs 1`] = `
<div>
  <TabsSection
    activeTabId="1"
    changeActiveFilterId={[Function]}
    columns={
      Array [
        Object {
          "displayName": "Inventory",
          "id": 1,
          "isDefault": false,
          "order": "first order",
        },
        Object {
          "displayName": "Price lists",
          "id": 2,
          "isDefault": true,
          "order": "second order",
        },
      ]
    }
    getAllItemsRequest={[Function]}
    getWithFilterRequest={[Function]}
    handleToggleSearchSection={[Function]}
    limit={10}
    offset={15}
    tabs={Array []}
    updateColumnsRequest={[Function]}
  />
</div>
`;
